File: app/cqrs/mr/readModel/ListView.java
package cqrs.mr.readModel;

import cqrs.core.Handler;
import cqrs.mr.events.InventoryItemCreated;
import cqrs.mr.events.InventoryItemDeactivated;
import cqrs.mr.events.InventoryItemRenamed;

public class ListView {

	public Handler<InventoryItemCreated> createInventoryItemCreatedHandler() {
		return new Handler<InventoryItemCreated>() {
			@Override
			public void handle(InventoryItemCreated message) {
				BullShitDatabase.list.add(new InventoryItemListDto(message.id(), message.name()));
			}
		};
	}

	public Handler<InventoryItemRenamed> createInventoryItemRenamedHandler() {
		return new Handler<InventoryItemRenamed>() {
			@Override
			public void handle(InventoryItemRenamed message) {
				for (InventoryItemListDto item : BullShitDatabase.list) {//simple solution, don't use it in production code!
					if (item.id.equals(message.id())) {
						item.name = message.newName();
						break;
					}
				}
			}
		};
	}

	public Handler<InventoryItemDeactivated> createInventoryItemDeactivatedHandler() {
		return new Handler<InventoryItemDeactivated>() {
			@Override
			public void handle(InventoryItemDeactivated message) {
				for (InventoryItemListDto item : BullShitDatabase.list) {//simple solution, don't use it in production code!
					if (item.id.equals(message.id())) {
						BullShitDatabase.list.remove(item);
						break;
					}
				}
			}
		};
	}
}

File: app/cqrs/mr/readModel/ListView.scala
package cqrs.mr.readModel

import cqrs.core.Handler
import cqrs.mr.events.{InventoryItemCreated, InventoryItemDeactivated, InventoryItemRenamed}

import scala.collection.JavaConverters._

object ListView {

  def createInventoryItemCreatedHandler(): Handler[InventoryItemCreated] = {
    (message: InventoryItemCreated) => {
      BullShitDatabase.list.add(new InventoryItemListDto(message.id, message.name))
    }
  }

  def createInventoryItemRenamedHandler(): Handler[InventoryItemRenamed] = {
    (message: InventoryItemRenamed) => {
      val item = BullShitDatabase.list.asScala.find(item => item.id == message.id).get
      item.name = message.newName
    }
  }

  def createInventoryItemDeactivatedHandler(): Handler[InventoryItemDeactivated] = {
    (message: InventoryItemDeactivated) => {
      val items = BullShitDatabase.list.asScala
      val pos = items.indexWhere(item => item.id == message.id)
      items.remove(pos)
    }
  }
}